mantidproject/mantid
# Remove ClassMacros.h file

Description of work.
This PR removes the `ClassMacros` file which provided a way of doing `MyConstructor() = delete;` before we had full compiler support. It fixes up all the files that previously used this.

I'm also benchmarking the build time before and after these changes out of curiosity to see what impact this change had on that:
Before: 154m 7s
After: 154m 2s

So very little impact on build time but a well needed tidy up anyway.

**To test:**
Ensure all tests pass
Code review - check that I have correctly swapped the **default constructor** and/or **assignment/copy** disable with the appropriate delete. 

Fixes #19824 .

*Does not need to be in the release notes. - Internal Change*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
